addons/purchase_stock/models/purchase_order.py

```diff
@@ -103,6 +103,12 @@ def write(self, vals):
                         to_log[order_line] = (order_line.product_qty, pre_order_line_qty[order_line])
                 if to_log:
                     order._log_decrease_ordered_quantity(to_log)
+        # Sync priority star state to linked pickings when updating an existing PO
+        if vals.get('priority'):
+            new_priority = vals['priority']
+            pickings_to_update = self.mapped('picking_ids').filtered(lambda p: p.priority != new_priority)
+            if pickings_to_update:
+                pickings_to_update.write({'priority': new_priority})
```

Comment:
```suggestion
         if 'priority' in vals:
            new_priority = vals['priority']
            self.picking_ids.filtered(lambda p: p.priority != new_priority).write({'priority': new_priority})
        return res
```

You don’t need to use `mapped` here since `picking_ids is a relational field`.
Also, in the `if condition`, there’s `no need to fetch the value from vals using get()`. We only need to check whether the `priority key exists in vals`.